### PR TITLE
fix: Make data as pointer to support module without default CR

### PR DESCRIPTION
**Pull request by ruanxin**
- configure moduletemplate `spec.data` as pointer so that generates empty field in case no default CR provided
- add e2e test to verify moduletemplate without default cr works
- upgrade go version in dockerfile
- refactor the existing test to better organize test utility code

bug reported in this issue: https://github.com/kyma-project/lifecycle-manager/issues/917

### CODE REVIEW

File: .github/workflows/test-e2e.yml [hunk 1]
```diff
@@ -164,6 +163,28 @@ jobs:
...
+        run: |
+          make build-manifests
+          echo "name: kyma-project.io/module/template-operator
+          channel: regular
+          version: v1.0.0
+          manifest: template-operator.yaml
+          security: sec-scanners-config.yaml
+          annotations:
+            operator.kyma-project.io/doc-url: https://kyma-project.io" >> module-config-no-cr.yaml
+          cat module-config-no-cr.yaml
```
Comment (Tomasz-Smelcerz-SAP) [inline .github/workflows/test-e2e.yml:178]: This is just for the debugging purposes? If so it should be removed.

File: .github/workflows/test-e2e.yml [hunk 2]
```diff
@@ -29,7 +29,7 @@ jobs:
           token: ${{ secrets.GITHUB_TOKEN }}
           # Context for which we should look for the matching status
           statusName: ${{ (github.event_name == 'pull_request') && 'pull-lifecycle-mgr-build' || 'main-lifecycle-mgr-build' }}
-          timeoutSeconds: 600
```
Comment (Tomasz-Smelcerz-SAP) [inline .github/workflows/test-e2e.yml:32]: How was this value obtained? Trial and error?

File: internal/controller/control-plane/helper_test.go [hunk 3]
```diff
@@ -112,6 +112,9 @@ func expectModuleTemplateSpecGetReset(
 	if err != nil {
 		return err
 	}
+	if moduleTemplate.Spec.Data == nil {
+		return ErrExpectedLabelNotReset
```
Comment (Tomasz-Smelcerz-SAP) [inline internal/controller/control-plane/helper_test.go:116]: Looks weird. Is that a correct error to return?

File: internal/controller/kyma_controller/kyma_test.go [hunk 4]
```diff
@@ -147,42 +149,41 @@ var _ = Describe("Kyma enable multiple modules", Ordered, func() {
...
+			Eventually(ManifestExists, Timeout, Interval).
+				WithContext(ctx).
+				WithArguments(controlPlaneClient, kyma.GetName(), kyma.GetNamespace(), activeModule.Name).
+				Should(Succeed())
 		}
 		By("Delete CR")
 		for _, activeModule := range kyma.Spec.Modules {
-			Eventually(deleteModule(kyma, activeModule), Timeout, Interval).Should(Succeed())
+			Eventually(DeleteModule, Timeout, Interval).
+				WithArguments(ctx, controlPlaneClient, kyma, activeModule.Name).Should(Succeed())
```
Comment (Tomasz-Smelcerz-SAP) [inline internal/controller/kyma_controller/kyma_test.go:168]: Use `WithContext(ctx)` as in other test cases.

File: .github/workflows/test-e2e.yml (hunk 2 above)
Comment (ruanxin) [reply .github/workflows/test-e2e.yml:32]: This is the timeout for waiting for image build, 10mins sometimes is not enough, you can find it in building history.

File: Makefile [hunk 5]
```diff
@@ -66,6 +66,7 @@ envtest-dir:
 .PHONY: test
 test: manifests test-crd generate fmt vet envtest ## Run tests.
 	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" go test `go list ./... | grep -v /tests/e2e_test` -coverprofile cover.out
+	go tool cover -html=cover.out
```
[the remark this answers is not in the document]
Comment (ruanxin) [reply Makefile:69]: this is mis committed, will remove it

File: internal/controller/control-plane/helper_test.go (hunk 3 above)
Comment (ruanxin) [reply internal/controller/control-plane/helper_test.go:116]: good catch, should be `ErrManifestResourceIsNil`

File: tests/e2e_test/status_propagation_test.go [hunk 6]
```diff
@@ -1,84 +1,74 @@
...
-			WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, v1beta2.StateWarning).
-			Should(Succeed())
+		It("Then module CR exist", func() {
+			Eventually(ModuleCRExists).
+				WithContext(ctx).
+				WithArguments(runtimeClient, moduleCR).
+				Should(Succeed())
+		})
+
+		It("Then resource defined in manifest CR", func() {
```
Comment (Tomasz-Smelcerz-SAP) [inline tests/e2e_test/status_propagation_test.go:36]: I am confused by the description of this case. Should it be: `Then resource [is] defined in manifest CR` or maybe: `Then resource defined in manifest CR [exists]`?

Comment (ruanxin) [reply tests/e2e_test/status_propagation_test.go]: changed to `Then resource is defined in manifest CR`

**Review by Tomasz-Smelcerz-SAP — APPROVED** [review]

**Pull request merged**